public/app/plugins/datasource/prometheus/language_utils.ts

```diff
@@ -1,8 +1,26 @@
 export const RATE_RANGES = ['1m', '5m', '10m', '30m', '1h'];
 
-export function processLabels(labels, withName = false) {
-  const values = {};
-  labels.forEach(l => {
+export const processHistogramLabels = (labels: string[]) => {
+  return labels.reduce(
```

Comment:
yes, instead of forEach use for of, but this is a reduce and so a bit different, might be better in some cases? The one thing to keep in mind is to never use map/reduce/forEach (callback based loops) if the array has many values (>500 or so) because for of loops are about 100 times faster than callback loops

Reply:
Thanks, are these agreements stated somewhere where I can read them?